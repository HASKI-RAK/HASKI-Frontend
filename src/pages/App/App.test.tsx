import { App } from "@pages";
import { render } from "@testing-library/react";
import renderer from "react-test-renderer";
import "@testing-library/jest-dom";

global.fetch = jest.fn(() =>
  Promise.resolve({
    json: () => Promise.resolve({ status: 200 }),
  }),
) as jest.Mock;

describe("Test the demo component", () => {
  test("renders correctly", () => {
    const tree = renderer.create(<App />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
