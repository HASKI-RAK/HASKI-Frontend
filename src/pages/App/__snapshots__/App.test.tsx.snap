// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test the demo component renders correctly 1`] = `
<div
  className="main"
>
  <div>
    Hello 
     
     with ID:
  </div>
  <button
    className="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium css-1u1dbmp-MuiButtonBase-root-MuiButton-root"
    disabled={false}
    onBlur={[Function]}
    onClick={[Function]}
    onContextMenu={[Function]}
    onDragLeave={[Function]}
    onFocus={[Function]}
    onKeyDown={[Function]}
    onKeyUp={[Function]}
    onMouseDown={[Function]}
    onMouseLeave={[Function]}
    onMouseUp={[Function]}
    onTouchEnd={[Function]}
    onTouchMove={[Function]}
    onTouchStart={[Function]}
    tabIndex={0}
    type="button"
  >
    one up
  </button>
  <div>
    <div
      className="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-colorPrimary LanguageDropdown css-jb90b-MuiInputBase-root-MuiOutlinedInput-root-MuiSelect-root"
      onClick={[Function]}
    >
      <div
        aria-expanded="false"
        aria-haspopup="listbox"
        className="MuiSelect-select MuiSelect-outlined MuiInputBase-input MuiOutlinedInput-input css-11u53oe-MuiSelect-select-MuiInputBase-input-MuiOutlinedInput-input"
        onBlur={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onMouseDown={[Function]}
        role="button"
        tabIndex={0}
      >
        Deutsch
      </div>
      <input
        aria-hidden={true}
        className="MuiSelect-nativeInput css-yf8vq0-MuiSelect-nativeInput"
        onAnimationStart={[Function]}
        onChange={[Function]}
        tabIndex={-1}
        value="de"
      />
      <svg
        aria-hidden={true}
        className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium MuiSelect-icon MuiSelect-iconOutlined css-hfutr2-MuiSvgIcon-root-MuiSelect-icon"
        data-testid="ArrowDropDownIcon"
        focusable="false"
        viewBox="0 0 24 24"
      >
        <path
          d="M7 10l5 5 5-5z"
        />
      </svg>
      <fieldset
        aria-hidden={true}
        className="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
      >
        <legend
          className="css-ihdtdm"
        >
          <span
            className="notranslate"
          >
            ​
          </span>
        </legend>
      </fieldset>
    </div>
  </div>
  <div>
    <div>
      Current Language: 
      de
       
      <br />
      Vorherige
       
      <br />
      Nächste
       
      <br />
      Hast du das gesucht?
       
      <br />
    </div>
  </div>
</div>
`;
